### DISCUSSION TITLE
How can I wait for a response from the user and put it in a variable

### DISCUSSION THREAD

**Discussion by GoladorYeah** [def-studio/telegraph#170] [Q&A] [answered]
Good day.
Perhaps a stupid question, but how can I wait for a response from the user and put it in a variable.
For example, when asked what is your name?

**Comment by fabio-ivona** [comment, marked as answer]
You should wait for a webhook, handling the user message, and keep an internal state of the conversation status

ps: conversations will come in v2, but we aim to a stable package first

**Reply by henzel** [reply]
May you give me a little example please?

I have webhook /start with button "Create task". Click on button -> send message with question about task title. What can I do later to save state? 

public function start() {
    $this->chat
    ->keyboard(Keyboard::make()->buttons([
        Button::make("Create task")->action("create_task_step_1"),
    ]))
    ->send();
}

public function create_task_step_1() {
    $text = 'Send task title;
    $this->chat->message($text)->send();
}

**Reply by fabio-ivona** [reply]
You could save a db entry with the user/bot/chat id and the status of the conversation
